xbmc/Util.cpp

```diff
@@ -1884,6 +1884,24 @@ void CUtil::ScanForExternalSubtitles(const std::string& strMovie, std::vector<st
   const std::vector<std::string> common_sub_dirs = { "subs", "subtitles", "vobsubs", "sub", "vobsub", "subtitle" };
   GetItemsToScan(strBasePath, g_advancedSettings.m_subtitlesExtensions, common_sub_dirs, items);
 
+  // Remove any rar archive assoiciated with the movie from the subtitle search
+  if(strMovie.find("rar://")!=std::string::npos){
+    int i = 0;
+    while(i < items.Size())
+      {
+	CFileItemPtr pItem = items[i];
+	std::string strTmp = URIUtils::GetFileName(pItem->GetPath());
+	URIUtils::RemoveExtension(strTmp);
+	std::size_t strip = strTmp.find_last_of(".part");
+	if(strip!=std::string::npos && strip>strTmp.length()-10)
```

Comment:
probably, many moons ago. this can be done a bit better though; use a CURL to extract the file name as it can be url encoded, possibly recursively (in case of embedded urls - rar in rar and such).